src/org/intellij/erlang/compilation/ErlangPrepareDependenciesCompileTask.java

```diff
@@ -220,67 +252,68 @@ private void addDeclaredDependencies(@NotNull ErlangFile erlangModule, @NotNull
-        @Nullable
-        @Override
-        public String fun(ErlangFile erlangFile) {
-          VirtualFile virtualFile = erlangFile.getVirtualFile();
-          return virtualFile != null && myPathsToDependenciesMap.containsKey(virtualFile.getPath()) ? virtualFile.getPath() : null;
-        }
-      });
+    private List<String> getDeclaredParseTransformPaths(@NotNull ErlangFile erlangModule) {
+      Set<String> pt = ContainerUtil.newHashSet();
+      erlangModule.addDeclaredParseTransforms(pt);
+      return resolvePathsFromNames(pt);
     }
 
     @NotNull
     private List<String> getDeclaredBehaviourPaths(@NotNull ErlangFile erlangModule) {
       Set<String> behaviours = ContainerUtil.newHashSet();
       ErlangPsiImplUtil.addDeclaredBehaviourModuleNames(erlangModule, behaviours);
-      return getPathsFromNames(behaviours);
+      return resolvePathsFromNames(behaviours);
     }
 
     @NotNull
-    private List<String> getDeclaredParseTransformPaths(@NotNull ErlangFile erlangModule) {
-      Set<String> pt = ContainerUtil.newHashSet();
-      erlangModule.addDeclaredParseTransforms(pt);
-      return getPathsFromNames(pt);
-    }
-
-    private void buildNamesMap(@NotNull Collection<VirtualFile> erlangModules) {
-      for (VirtualFile erlangModule : erlangModules) {
-        myNamesToPathsMap.put(erlangModule.getNameWithoutExtension(), erlangModule.getPath());
+    private List<String> resolvePathsFromNames(@NotNull Collection<String> erlangModuleNames) {
+      List<String> paths = ContainerUtil.newArrayList();
+      for (String erlangModuleName : erlangModuleNames) {
+        paths.addAll(getPathsFromModuleName(erlangModuleName));
       }
+      return paths;
     }
 
     @NotNull
-    private List<String> getPathsFromNames(@NotNull Collection<String> erlangModuleNames) {
-      return ContainerUtil.mapNotNull(erlangModuleNames, new Function<String, String>() {
+    private List<String> getDeclaredIncludePaths(@NotNull ErlangFile file) {
+      return ContainerUtil.mapNotNull(ErlangPsiImplUtil.getDirectlyIncludedFiles(file), new Function<ErlangFile, String>() {
+        @Nullable
         @Override
-        public String fun(String name) {
-          return myNamesToPathsMap.get(name);
+        public String fun(ErlangFile erlangFile) {
+          VirtualFile virtualFile = erlangFile.getVirtualFile();
+          return virtualFile != null && myHeaders.contains(virtualFile.getPath()) ? virtualFile.getPath() : null;
         }
       });
     }
 
-    @Override
-    public Collection<String> getNodes() {
-      return myPathsToDependenciesMap.keySet();
+    @NotNull
+    private List<String> getPathsFromModuleName(@NotNull String erlangModuleName) {
+      return ContainerUtil.mapNotNull(ErlangModuleIndex.getFilesByName(myProject, erlangModuleName, GlobalSearchScope.projectScope(myProject)), new Function<ErlangFile, String>() {
```

Comment:
Incorrect scope - I believe, it should be module with dependencies, or something.

Also, could you, please make this line fit the screen?